lib/delayed/tasks.rb

```diff
@@ -30,7 +30,7 @@
   task :check, [:max_age] => :environment do |_, args|
     args.with_defaults(:max_age => 300)
 
-    unprocessed_jobs = Delayed::Job.where('attempts = 0 AND created_at < ?', Time.now - args[:max_age].to_i).count
+    unprocessed_jobs = Delayed::Job.where('attempts = 0 AND created_at < ? AND run_at <= ?', Time.now - args[:max_age].to_i, Time.now).count
```

Comment:
Conceptually I think the max age applies to both times. Probably:
```suggestion
    unprocessed_jobs = Delayed::Job.where('attempts = 0 AND created_at < :time AND run_at < :time', time: args[:max_age].to_i.seconds.ago).count
```

And it is possibly better to replace `created_at` with `run_at` instead of querying both.